Sources/ElevenLabs/Public/Conversation/Conversation.swift

```diff
@@ -0,0 +1,774 @@
+        if let mode = config?.microphoneMuteMode {
+            try? audioManager.set(microphoneMuteMode: mode)
+        }
+
+        if let bypass = config?.voiceProcessingBypassed {
+            audioManager.isVoiceProcessingBypassed = bypass
+        }
+
+        if let agc = config?.voiceProcessingAGCEnabled {
+            audioManager.isVoiceProcessingAGCEnabled = agc
+        }
+
+        if let prepared = config?.recordingAlwaysPrepared {
+            try? await audioManager.setRecordingAlwaysPreparedMode(prepared)
+        }
+
+        let needsSpeechHandler = (config?.onSpeechActivity != nil) || (options.onSpeechActivity != nil)
+
+        if needsSpeechHandler {
+            if !audioSpeechHandlerInstalled {
+                previousSpeechActivityHandler = audioManager.onMutedSpeechActivity
+                audioSpeechHandlerInstalled = true
+            }
+            audioManager.onMutedSpeechActivity = { [weak self] _, event in
+                guard let self else { return }
+                Task { @MainActor in
+                    // Handlers are @Sendable, they manage their own synchronization
+                    if let handler = self.options.audioConfiguration?.onSpeechActivity {
+                        handler(event)
+                    }
+                    if let handler = self.options.onSpeechActivity {
+                        handler(event)
+                    }
+                }
+            }
+        } else if audioSpeechHandlerInstalled {
+            audioManager.onMutedSpeechActivity = previousSpeechActivityHandler
+            previousSpeechActivityHandler = nil
+            audioSpeechHandlerInstalled = false
+        }
+    }
+
+    private func normalizeConversationError(
+        _ error: Error,
+        default defaultError: (Error) -> ConversationError
+    ) -> ConversationError {
+        if let conversationError = error as? ConversationError {
+            return conversationError
+        }
+        return defaultError(error)
+    }
+
+    private func updateStartupState(_ newState: ConversationStartupState) {
+        startupState = newState
+        options.onStartupStateChange?(newState)
+    }
+
+    private func resetFlags() {
+        // Don't reset isMuted - it should reflect actual room state
+        agentState = .listening
+        pendingToolCalls.removeAll()
+        mcpToolCalls.removeAll()
+        mcpConnectionStatus = nil
+        conversationMetadata = nil
+        startupMetrics = nil
+        latestAudioAlignment = nil
+        latestAudioEvent = nil
+    }
+
+    /// Clean up state from any previous conversation to ensure a fresh start.
+    /// This method ensures that each new conversation starts with a clean slate,
+    /// preventing any state leakage between conversations when using new Room objects.
+    private func cleanupPreviousConversation() {
+        // Cancel any ongoing tasks and reset references immediately
```

Comment:
Given this is called when starting a new conversation, if it lags is it possible these changes would effect the new conversation?